subprojects/docs/src/docs/userguide/authoring-builds/writing_build_scripts.adoc

```diff
@@ -200,87 +211,119 @@ include::{snippetsPath}/tutorial/configureObjectUsingScript/tests/configureObjec
 ====
 
 [[groovy-dsl-basics]]
-== Some Groovy basics
-
-[TIP]
-Looking for some Kotlin basics, the link:https://kotlinlang.org/docs/reference/[Kotlin reference documentation] and link:https://kotlinlang.org/docs/tutorials/koans.html[Kotlin Koans] should be useful to you.
+== Helpful Groovy Syntax
 
-The https://docs.groovy-lang.org/latest/html/documentation/index.html[Groovy language] provides plenty of features for creating DSLs, and the Gradle build language takes advantage of these. Understanding how the build language works will help you when you write your build script, and in particular, when you start to write custom plugins and tasks.
+Groovy provides many features to help create DSLs. These features can help you write better build scripts, plugins, and tasks.
 
+[TIP]
+Looking for helpful Kotlin syntax? Check out the link:https://kotlinlang.org/docs/reference/[Kotlin reference documentation] and link:https://kotlinlang.org/docs/tutorials/koans.html[Kotlin Koans].
 
 [[sec:groovy_jdk]]
-=== Groovy JDK
+=== Convenience Methods
 
-Groovy adds lots of useful methods to the standard Java classes. For example, `Iterable` gets an `each` method, which iterates over the elements of the `Iterable`:
+Groovy adds convenience methods to the standard Java classes.
+For example, `Iterable` gets an `each` method, which iterates over the elements of the `Iterable`:
```

Comment:
I think this is a bad example of an extension - it's too close to the API and functionality already available in the JDK.  Maybe something like `File.text`, which hides a lot of the difficult details of file IO that Java devs should be very familiar with.

Reply:
The thing about something like File.text is that it's really File.setText() and getText(). We shouldn't introduce two Groovy features things at once here. I think each is a good example.

Reply:
That's true.  I just think that with `.forEach()` in JDK 8, and some form of callback-based iteration basically expected as a language feature anywhere today, `.each()` is not such a good example.  Maybe `List.head()` and `List.tail()`?

Reply:
It sounds like there are mixed pros and cons for changing this example. Would it be OK to put this code block update on the backburner for now, since the current example is workable? I can revise it later.